src/kernel/src/discord_presence/mod.rs

```diff
@@ -0,0 +1,57 @@
+use crate::{info, warn};
+use discord_rich_presence::{activity, DiscordIpc, DiscordIpcClient};
+use param::Param;
+use std::fs::File;
+use std::path::Path;
+use std::time::SystemTime;
+
+pub fn rich_presence(game_path: &Path) -> Result<DiscordIpcClient, Box<dyn std::error::Error>> {
+    info!("Initializing Discord rich presence.");
+    let mut client = match DiscordIpcClient::new("1168617561244565584") {
+        Ok(client) => client,
+        Err(e) => {
+            warn!("Failed to create Discord IPC: {:?}", e);
+            return Err(e.into());
+        }
+    };
+
+    if let Err(e) = client.connect() {
+        warn!("Failed to connect to Discord client: {:?}", e);
+        return Err(e.into());
+    }
+
+    let start_time: i64 = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)?
+        .as_secs()
+        .try_into()
+        .unwrap();
+    let param_path = game_path.join("sce_sys/param.sfo");
```

Comment:
The correct way to do this is separating `sce_sys` and `param.sfo` into each join.